events/sse_events_test.go

```diff
@@ -0,0 +1,154 @@
+package events
+
+import (
+	"bufio"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEvent_IfEventIsEmptyReturnsFalse(t *testing.T) {
+	t.Parallel()
+	// given
+	event := &SSEEvent{Timestamp: Timestamp{},
+		Type: "status_update_event",
+		Body: []byte(`{"id": "simpleId"}`),
+		ID:   "id",
+	}
+	// when
+	expected := false
+	actual := event.isEmpty()
+	// then
+	assert.Equal(t, expected, actual)
+}
+
+func TestEvent_IfEventIsEmptyReturnsTrue(t *testing.T) {
+	t.Parallel()
+	// given
+	event := &SSEEvent{}
+	// when
+	expected := true
+	actual := event.isEmpty()
+	// then
+	assert.Equal(t, expected, actual)
+}
+
+func TestParseLine_WhenStautsUpdateEventPassed(t *testing.T) {
+	t.Parallel()
+	// given
+	event := &SSEEvent{}
+	line0 := []byte("id: 0")
+	line1 := []byte("event: status_update_event")
+	line2 := []byte("data: testData")
+	expected0 := "0"
+	expected1 := "status_update_event"
+	expected2 := []byte("testData\n")
+	// when
+	event.parseLine(line0)
+	event.parseLine(line1)
+	event.parseLine(line2)
+	// then
+	assert.Equal(t, expected0, event.ID)
+	assert.Equal(t, expected1, event.Type)
+	assert.Equal(t, string(expected2), string(event.Body))
+}
+
+func TestParseLine_WhenGarbageIsProvidedBodyShouldBeNil(t *testing.T) {
+	t.Parallel()
+	// given
+	event := &SSEEvent{}
+	line := []byte("garbage data")
+	expectedBody := []byte(nil)
+	// when
+	_ = event.parseLine(line)
+	// then
+	assert.Equal(t, expectedBody, event.Body)
+}
+
+// given
+var parseEventCases = []struct {
+	in        string
+	eventType string
+	eventBody []byte
+}{
+	{"event: status_update_event\ndata: testData\n", "status_update_event", []byte("testData\n")},
+	{"event: status_update_event\ndata: testData\ndummydata", "status_update_event", []byte("testData\n")},
+	{"event: status_update_event\ndata", "status_update_event", nil},
+	{"event: status_update_event\ndata:\n", "status_update_event", nil},
+}
+
+func TestParseSSEEvent_TestCases(t *testing.T) {
```

Comment:
Missing test for multiline data, unsupported fields, comments. You can copy it from https://github.com/allegro/marathon-consul/pull/144/files#diff-18e82633d2b99b89eb712771b083a18eR44